src/sentry/lang/dart/utils.py

```diff
@@ -29,36 +29,39 @@ def get_debug_meta_image_ids(event: dict[str, Any]) -> set[str]:
     }
 
 
-def generate_dart_symbols_map(debug_id: str, project: Project):
+def generate_dart_symbols_map(debug_ids: list[str], project: Project):
     """
-    Fetches and returns the dart symbols mapping for the given debug_id.
+    Fetches and returns the dart symbols mapping for the first available debug_id.
+    There should only be one mapping file per Flutter build, so we return the first mapping found.
     """
     with sentry_sdk.start_span(op="dartsymbolmap.generate_dart_symbols_map") as span:
-        try:
-            dif_paths = ProjectDebugFile.difcache.fetch_difs(
-                project, [debug_id], features=["mapping"]
-            )
-            debug_file_path = dif_paths.get(debug_id)
-            if debug_file_path is None:
-                return
-
-            dart_symbols_file_size_in_mb = os.path.getsize(debug_file_path) / (1024 * 1024.0)
-            span.set_tag("dartsymbolmap_file_size_in_mb", dart_symbols_file_size_in_mb)
-
-            with open(debug_file_path, "rb") as f:
-                data = orjson.loads(f.read())
-
-            if isinstance(data, list):
-                # Array format - transform it to map
-                if len(data) % 2 != 0:
-                    raise Exception("Debug array contains an odd number of elements")
-                # Obfuscated names are the odd indices and deobfuscated names are the even indices
-                return dict(zip(data[1::2], data[::2]))
-            else:
-                raise Exception(f"Unexpected dartsymbolmap format: {type(data)}")
-        except Exception as err:
-            sentry_sdk.capture_exception(err)
-            return
+        for debug_id in debug_ids:
```

Comment:
You don't need to loop at all here, you can just give `fetch_difs` a list of debug IDs.

Reply:
thx, updated it with `fetch_difs` and then without loops use the first debug file path of the result to create the map